scripts/datacube.py

```diff
@@ -394,22 +403,63 @@ def process(
     return ds_merge
 
 
-def main():
-    tiles = gpd.read_file("scripts/data/mgrs_sample.geojson")
-    sample = tiles.sample(1, random_state=45)
-    aoi = sample.iloc[0].geometry
+def convert_attrs_and_coords_objects_to_str(data):
+    """
+    Convert attributes and coordinates that are objects to
+    strings.
+
+    This is required for storing the xarray in netcdf.
+    """
+    for key, coord in data.coords.items():
+        if coord.dtype == "object":
+            data.coords[key] = str(coord.values)
+
+    for key, attr in data.attrs.items():
+        data.attrs[key] = str(attr)
+
+    for key, var in data.variables.items():
+        var.attrs = {}
+
+
+@click.command()
+@click.option(
+    "--index",
+    required=False,
+    default=23,
+)
+def main(index):
+    print("Starting algorithm", index)
+    index = int(index)
+    tiles = gpd.read_file("mgrs_sample.geojson")
+    tile = tiles.iloc[index]
     start_year = 2017
     end_year = 2023
-
     merged = process(
-        aoi,
+        tile.geometry,
         start_year,
         end_year,
         SPATIAL_RESOLUTION,
         CLOUD_COVER_PERCENTAGE,
         NODATA_PIXEL_PERCENTAGE,
     )
-    return merged
+    date = np.datetime_as_string(merged.time.values)[:10]
+    mgrs = tile["name"]
+    version = "0.1"
+
+    merged = merged.compute()
+
+    convert_attrs_and_coords_objects_to_str(merged)
+    encoding = {key: {"zlib": True, "complevel": 9} for key in merged.variables.keys()}
+
+    with tempfile.NamedTemporaryFile() as dat:
```

Comment:
Not critical, but we can add a suffix of `.nc` to the temporary file.
```suggestion
    with tempfile.NamedTemporaryFile(suffix=".nc") as dat:
```

Reply:
sounds good too I will change this for the tiling and then use the suffix for the tiles